docs/topics/patterns/instant-messaging.mdx

```diff
@@ -0,0 +1,359 @@
+
+:::tip
+To store the time as the score, you must use an integer representation of the time. Store either ticks or time in [milliseconds since the Unix epoch](https://www.epochconverter.com/).
+:::
+
+Sorted sets have a [rich set of features](../../cache/develop/api-reference/sorted-set-collections) that allow you to access messages in many different ways. Consider the following use cases:
+
+* Paging messages N at a time
+* Retrieving messages in a certain time frame
+* Displaying messages in reverse order
+* Getting all messages ever sent
+
+Based on the parameters you use in your fetch command, you can easily satisfy any of the use cases above.
+
+A significant point to remember - **messages must be unique in sorted sets**! This means when saving the message to the sorted set, you should include some sort of message identifier or timestamp in your JSON.
+
+#### List
+
+A [list](../../cache/develop/basics/datatypes#lists) is a collection data type that stores an ordered array of elements. You can push elements to either the front or back of a list and you can add multiple entries at one time. A list is a great data type for instant messaging because it stores messages in the order they were sent. When users join an instant messaging session, you can quickly fetch the entire chat history with a single [listFetch API call](../../cache/develop/api-reference/list-collections#listfetch).
+
+Elements in a list do not need to be unique. You can push the same message into a list over and over again and it will be added every time. Lists are a good choice for instant messaging for the following use cases:
+
+* Paging messages N at a time
+* Adding multiple messages at once
+* Maintaining a chat history of fixed length, i.e. the last 100 messages
+
+When using a list, you have more array manipulation functions than with a sorted set, allowing you to automatically truncate message history if it gets too long or remove specific elements for things like content moderation.
+
+### Delivery
+
+Technically, message storage is not a necessary component for instant messaging. In theory you could have a messaging system that only showed the messages that were sent while a user is connected. Messages could be stored in memory in the client and you'd be done. That said, message delivery is a mandatory part of instant messaging. Getting the message from the sender to all recipients with minimal latency is the core of this pattern. To do this, we use [Momento Topics](https://www.gomomento.com/platform/topics).
+
+Momento Topics is a low-latency pub/sub service that connects clients together, clients to servers, servers to clients, and servers to servers. Think WebSockets but *way better*. When you publish a message to a topic, Momento will deliver it to all subscribers of the topic, providing fan-out capability or 1 to 1, depending on your use case.
```

Comment:
might be nice to get the word `broadcast` in here? maybe `Momento will broadcast`?